eos-metapackage

```diff
@@ -71,16 +81,52 @@ sub get_depends {
 	return @deps;
 }
 
+sub check_installable {
+	my ($depfile, $deps) = @_;
+	my @cmd;
+	my %options = (
+		# ignore packages for foreign architectures; they may not be
+		# installable on this architecture, and that's OK.
+		reduce_restrictions => 1,
+	);
+
+	verbose_print("checking all packages in $depfile are installable");
+	deps_iterate(deps_parse($deps, %options), sub {
```

Comment:
Fancy

Reply:
It's actually the easiest way to iterate this thing, AFAIK, because `deps_parse` always gives you back a collection of deps, even if it's a degenerate AND of one dep. It would also cope if we ever did introduce conditional deps `x | y`.